pkg/shell/hosts.jsx

```diff
@@ -0,0 +1,232 @@
+import cockpit from "cockpit";
+
+import React from 'react';
+import ReactDOM from "react-dom";
+import PropTypes from 'prop-types';
+import { PageSidebar, Button } from '@patternfly/react-core';
+import { EditIcon, TrashIcon } from '@patternfly/react-icons';
+
+import { CockpitNav, CockpitNavItem } from "./nav.jsx";
+
+import { machines } from "machines";
+import { new_machine_dialog_manager } from "machine-dialogs";
+
+import "../../node_modules/@patternfly/patternfly/components/Select/select.scss";
+
+import $ from "jquery"; // Ech... Fixme: We now use the same dialogs as dashboard and we need jQuery to open them
+
+const _ = cockpit.gettext;
+const hosts_sel = document.getElementById("hosts-nav");
+
+class HostsSelector extends React.Component {
+    constructor() {
+        super();
+        this.el = document.createElement("div");
+        this.el.className = "view-hosts";
+    }
+
+    componentDidMount() {
+        hosts_sel.appendChild(this.el);
+    }
+
+    componentWillUnmount() {
+        hosts_sel.removeChild(this.el);
+    }
+
+    render() {
+        const { children } = this.props;
+        return ReactDOM.createPortal(children, this.el);
+    }
+}
+
+function HostLine(props) {
+    const host = props.host;
+    const user = props.user;
```

Comment:
Nitpick, but please enumerate props explicitly for functional components, it makes it easier to see what props they expect:

    const HostLine({ host, user }) => (
        <dom stuff>
    );